src/main/kotlin/no/nav/amt/person/service/nav_ansatt/NavAnsattUpdater.kt

```diff
@@ -0,0 +1,25 @@
+package no.nav.amt.person.service.nav_ansatt
+
+import no.nav.amt.person.service.clients.nom.NomClient
+import org.springframework.stereotype.Component
+
+@Component
+class NavAnsattUpdater(
+	private val nomClient: NomClient,
+	private val navAnsattRepository: NavAnsattRepository,
+) {
+
+	private val batchSize = 100
+
+	fun oppdaterAlle() {
+		val ansattBatcher = navAnsattRepository.getAll()
+			.map { it.navIdent }
+			.chunked(batchSize)
+
+		ansattBatcher.forEach { batch ->
+			val oppdaterteAnsatte = nomClient.hentNavAnsatte(batch)
+			navAnsattRepository.updateMany(oppdaterteAnsatte)
```

Comment:
Jeg er litt usikker på om det er lurt å bruke `NomNavAnsatt` direkte i repositoryet. Men jeg vet ikke helt hvorfor ikke i såfalll.

I  `amt-tiltak` så hadde vi en `log.warn` her om vi ikke fant den navansatte som skulle oppdateres. Det Blir litt mer kronglete med en batch, men vi kan mappe resultatet tilbake til de ansatte i batchen for å sjekke. Men jeg finner ikke spor av den loggen de siste 30 dagene, så jeg vet ikke hvor viktig det er å ha.

Reply:
Det er vel egentlig god praksis å ikke bruke et objekt som tilhører et eksternt grensesnitt internt i appen. Det er vel av hensyn til anti corruption slik at hvis grensesnittet endrer seg så slipper du å endre i hele appen din, inkludert databasen. Men, dette er jo en ganske enkelt grensesnitt, og vi plukker vel bare de feltene vi har behov for fra responsen uansett, så da tror jeg ikke det skader å bruke det samme objektet hvis du synes det blir enklere (koden blir jo ikke alltid så lett å vedlikeholde hvis man har enn drøss med helt like objekter heller 😅 )

Reply:
Hva betyr det egentlig at vi ikke finner den nav-ansatte? Betyr det at vedkommende har sluttet? I så fall bør vi jo kanskje heller slette den. Hvis det ikke er altfor vanskelig å logge når vi ikke kan oppdatere er jo det fint, men det er ikke verdens undergang om vi ikke får det til heller. Siden du oppdaterer timestamp for når den sist ble oppdatert er det forsåvidt mulig å fange opp hvis noen aldri oppdateres sånn sett også.

Reply:
Jeg vet ikke ikke hva det betyr når vi ikke finner den selv. Jeg tror ikke vi har hatt noen rutine for å slette ansatte, eller fange opp at noen har sluttet.

Jeg tenker kanskje det er lurest å mappe resultatet fra Nom tilbake til de ansatte i batchen igjen, siden det løser egentlig begge problemene mine, og jeg tror ikke det har noe å si om det går litt tregere uansett.